src/Authenticator/LoginHandler.php

```diff
@@ -155,9 +209,52 @@ public function startRegister(HTTPRequest $request)
      * @param HTTPRequest $request
      * @return HTTPResponse
      */
-    public function finishRegister(HTTPRequest $request)
+    public function finishRegistration(HTTPRequest $request)
     {
-        return $this->jsonResponse(['errors' => ['Registration not yet implemented']], 500);
+        $sessionStore = $this->getSessionStore();
+        $sessionMember = $sessionStore->getMember();
+        $loggedInMember = Security::getCurrentUser();
+
+        if (is_null($loggedInMember) && is_null($sessionMember)) {
+            return $this->jsonResponse(
+                ['errors' => [_t(__CLASS__ . '.NOT_AUTHENTICATING', 'You must be logged or logging in')]],
+                403
+            );
+        }
+
+        $storedMethodName = $sessionStore->getMethod();
+        $specifiedMethodName = str_replace('-', '\\', $request->param('Method'));
```

Comment:
Can we centralise this logic (since I've seen it twice) somewhere? Maybe a service or utility class of some sort?

Reply:
We now have the `getURLSegment` method that we're using in place of this.

Reply:
It might make sense to add a method to `MethodRegistry` for consuming this... `getMethodFromURLSegment(string $segment): MethodInterface`.